test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -848,6 +987,58 @@ public void CreateValidationDetails_ValidatableObject_AlreadyInContext_Ignores()
             Assert.Same(attribute, validatorMetadata);
         }
 
+        private IEnumerable<KeyValuePair<EnumGroupAndName, string>> GetLocalizedEnumGroupedDisplayNamesAndValues(
+            bool useStringLocalizer)
+        {
+            var provider = CreateIStringLocalizerProvider(useStringLocalizer);
+
+            var key = ModelMetadataIdentity.ForType(typeof(EnumWithLocalizedDisplayNames));
+            var attributes = new object[0];
+
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+            provider.CreateDisplayMetadata(context);
+
+            return context.DisplayMetadata.EnumGroupedDisplayNamesAndValues;
+        }
+
+        private DataAnnotationsMetadataProvider CreateIStringLocalizerProvider(bool useStringLocalizer)
+        {
+            var stringLocalizer = new Mock<IStringLocalizer>(MockBehavior.Strict);
+            stringLocalizer
+                .Setup(loc => loc[It.IsAny<string>()])
+                .Returns<string>((k =>
+                {
+                    return new LocalizedString(k, $"{k} {CultureInfo.CurrentCulture}");
+                }));
+
+            var stringLocalizerFactory = new Mock<IStringLocalizerFactory>(MockBehavior.Strict);
+            stringLocalizerFactory
+                .Setup(factory => factory.Create(typeof(EnumWithLocalizedDisplayNames)))
+                .Returns(stringLocalizer.Object);
+
+            return new DataAnnotationsMetadataProvider(
+                useStringLocalizer ? stringLocalizerFactory.Object : null);
+        }
+
+        private class KeyValuePairEnumGroupAndName : IEqualityComparer<KeyValuePair<EnumGroupAndName, string>>
+        {
+            public bool Equals(KeyValuePair<EnumGroupAndName, string> x, KeyValuePair<EnumGroupAndName, string> y)
+            {
+                return x.Key.Name.Equals(y.Key.Name, StringComparison.Ordinal)
+                    && x.Key.Group.Equals(y.Key.Group, StringComparison.Ordinal);
```

Comment:
Do this w/in a `using (new CultureReplacer(string.Empty, string.Empty))` to play fair. Same for `GetHashCode()`.